internal/common/tools/tools.go

```diff
@@ -73,19 +74,15 @@ func IncrementIP(origIP, cidr string) (string, error) {
 	return ip.String(), nil
 }
 
-func gnbIdGenerator(i int) string {
+func gnbIdGenerator(i int, gnbId string) string {
 
-	var base string
-	switch true {
-	case i < 10:
-		base = "00000"
-	case i < 100:
-		base = "0000"
-	case i >= 100:
-		base = "000"
+	gnbId_int, err := strconv.Atoi(gnbId)
+	if err != nil {
+		log.Fatal("[UE][CONFIG] Given gnbId is invalid")
 	}
+	base := gnbId_int + i
 
-	gnbId := base + strconv.Itoa(i)
+	gnbId = fmt.Sprintf("%06d", base)
```

Comment:
I know the codebase currently only supports 24-bit gNB ID. This should be generalized in the future, to allow configuration length of gNB ID vs Cell ID.
https://www.techplayon.com/5g-nr-cell-global-identity-planning/

Reply:
To be implented in https://github.com/HewlettPackard/PacketRusher/issues/51